pkg/registry/core/service/rest.go

```diff
@@ -350,6 +350,37 @@ func (rs *REST) Update(ctx genericapirequest.Context, name string, objInfo rest.
 		return nil, false, errors.NewConflict(api.Resource("services"), service.Namespace, fmt.Errorf("Service.Namespace does not match the provided context"))
 	}
 
+	// TODO: this should probably move to strategy.PrepareForCreate()
```

Comment:
Thanks Harry! In fact, most of the code here is copy & paste from [func Create](https://github.com/xiangpengzhao/kubernetes/blob/e10f6e0f99b5f34dea48cddeb964a7b294c54ab7/pkg/registry/core/service/rest.go#L80), so I leave the existing TODO as is.